Smooth Line View/SmoothLineView.m

```diff
@@ -138,29 +127,35 @@ - (void) drawRect:(CGRect)rect {
     [self.bgColor set];
     UIRectFill(rect);
     
-    // get the graphics context and draw the path
-    CGContextRef context = UIGraphicsGetCurrentContext();
-    CGContextAddPath(context, _path);
-    CGContextSetLineCap(context, kCGLineCapRound);
-    CGContextSetLineWidth(context, self.lineWidth);
-    CGContextSetStrokeColorWithColor(context, self.lineColor.CGColor);
-    
-    CGContextStrokePath(context);
-    
-    if (self.renderAsArea)
+    @synchronized (self)
     {
-        CGContextAddPath(context, _path);
-        CGContextSetFillColorWithColor(context, [UIColor redColor].CGColor);
-        CGContextSetAlpha(context, 0.2);
-        CGContextFillPath(context);
+        // get the graphics context and draw the path
+        CGContextRef context = UIGraphicsGetCurrentContext();
```

Comment:
Get context and use context to get clip rect.

Use `drawLayer` and leave `drawRect` empty.